test/integration/main_test.go

```diff
@@ -358,6 +358,149 @@ deletion_timeline:
+// into bigquery.
+func TestMappingReusableWorkflowCall(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name                string
+		bigqueryTable       string
+		traceID             string
+		wantResourceMapping *v1alpha1.ResourceMapping
+	}{
+		{
+			name:          "mapping-copy-success-event",
+			bigqueryTable: cfg.MappingTableID,
+			traceID:       "fake-pmap-dev-manual-test-mapping",
+			wantResourceMapping: &v1alpha1.ResourceMapping{
+				Resource: &v1alpha1.Resource{
+					Provider: "gcp",
+					Name:     fmt.Sprintf("//storage.googleapis.com/%s", "pmap-static-ci-bucket-7faa"),
+				},
+				Contacts: &v1alpha1.Contacts{Email: []string{"pmap.mapping@gmail.com"}},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			t.Logf("using trace ID %s", tc.traceID)
+			t.Logf("using workflow run ID %s", cfg.WorkflowRunID)
+			// Check if the file uploaded exists in BigQuery.
+			queryString := fmt.Sprintf("SELECT * FROM `%s.%s.%s`", cfg.ProjectID, cfg.BigQueryDataSetID, tc.bigqueryTable)
+			queryString += `WHERE JSON_VALUE(data.githubSource.workflowRunId) = ?`
+
+			bqQuery := bqClient.Query(queryString)
+			bqQuery.Parameters = []bigquery.QueryParameter{{Value: cfg.WorkflowRunID}}
+
+			gotBQEntry := testGetFirstMatchedBQEntryWithRetries(ctx, t, bqQuery, cfg)
+
+			gotPmapEvent := &v1alpha1.PmapEvent{}
+			if err := protojson.Unmarshal([]byte(gotBQEntry.Data), gotPmapEvent); err != nil {
+				t.Fatalf("failed to unmarshal BQEntry.Data to pmapevent: %v", err)
+			}
+
+			gotResourceMapping := &v1alpha1.ResourceMapping{}
+			if err := gotPmapEvent.GetPayload().UnmarshalTo(gotResourceMapping); err != nil {
+				t.Fatalf("failed to unmarshal to resource mapping: %v", err)
+			}
+
+			cmpOpts := []cmp.Option{
+				protocmp.Transform(),
+				protocmp.IgnoreFields(&v1alpha1.ResourceMapping{}, "annotations"),
+			}
+
+			if diff := cmp.Diff(tc.wantResourceMapping, gotResourceMapping, cmpOpts...); diff != "" {
+				t.Errorf("gotPayload unexpected diff (-want,+got):\n%s", diff)
+			}
+
+			if diff := cmp.Diff(tc.traceID, gotResourceMapping.Annotations.AsMap()["traceID"]); diff != "" {
+				t.Errorf("traceID got unexpected diff (-want,+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+// TestMappingReusableWorkflowCall tests the policy file-copy reusable workflow has successfully
+// uploaded the file to gcs and triggered pmap event handler, and write the corresponding entry
+// into bigquery.
+func TestPolicyReusableWorkflowCall(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name          string
+		bigqueryTable string
+		traceID       string
+		wantPayload   *structpb.Struct
+	}{
+		{
+			name:          "policy-copy-success-event",
+			bigqueryTable: cfg.PolicyTableID,
+			traceID:       "fake-pmap-dev-manual-test-policy",
+			wantPayload: &structpb.Struct{
```

Comment:
Don't use resource mapping example for policy

Reply:
Updated.